### ISSUE TITLE
Question on zero-shot setting

### ISSUE THREAD

**Issue by wnn2000** [trishullab/escher#2]
Hi, thanks for your good work.

I'd like to know how to use your method in the zero-shot setting (section 5.3). In my understanding, your method need data to update the concept set. But in the zero-shot setting, there is no training data, how is it achieved?

**Comment by atharvas** [comment]
Hello! Thank you for reading the paper!

I wrote a step-by-step explanation of how the zero-shot iteration works here: https://trishullab.github.io/escher-web/.  In the zero-shot setting, Escher uses the CLIP model's confidence score to generate feedback to update the concept set.

Let me know if you have any follow up questions!

**Comment by wnn2000** [comment]
Thanks for your reply. However, I did not find the content regarding zero-shot iteration in your provided page. If possible, could you give me more details on using the CLIP model's confidence score to generate feedback?

**Comment by atharvas** [comment]
Of course. The [escher/models/model_zero_shot.py](https://github.com/trishullab/escher/blob/main/escher/models/model_zero_shot.py) file might also be useful to read. I don't remember the exact details off the top of my head but here is a summary:

1. Start with an LLM generated dictionary mapping each class to a set of concepts.
2. Collect the CLIP model's confidence score for a large set of unlabelled images and a large number of concepts (produces tensor of dimensions `[n_imgs, n_concepts]`). 
3. aggregate concepts based on the LLM-generated dictionary to get `[n_imgs, n_classes]`. 
4. aggregate the top-k classes (`[n_imgs, 3]` for `topk=3`)
5. calculate a confusion matrix (`[n_classes, n_classes]` where each entry is the number of times clsA and clsB were in the top-k classes).
6. This confusion matrix allows us to edit the initial LLM generated dictionary mapping.

Step 4 and 5 can be changed out for other strategies of converting `[n_imgs, n_classes] -> [n_classes, n_classes]`.

**Comment by wnn2000** [comment]
Got it. Thanks for your reply. But I am concerned that this setting is not very aligned with the term of ''zero-shot'' in our community.

**Comment by atharvas** [comment]
Hi! Apologies for the delay in getting back to you!  No, the [CBM we are optimizing](
https://arxiv.org/pdf/2210.07183) is a zero-shot algorithm. As mentioned in the paper and the blog post, Escher is a meta-algorithm that aims to enhance the performance of CBMs that operate in the zero-shot, few-shot, and fully finetuned setting without access to the ground truth labels. If there are any further misunderstandings, feel free to reach out to me at `atharvas@utexas.edu` -- I prefer keeping the github issues for code related issues but happy to talk more about this via email!
